distributed/utils.py

```diff
@@ -1184,7 +1184,7 @@ def get_event_loop(self):
         asyncio.set_event_loop_policy(AnyThreadEventLoopPolicy())
 
 
-@functools.lru_cache(1000)
+@functools.cache
```

Comment:
```suggestion
@functools.lru_cache
```

Reply:
Thoughts on why the default 128 is better than no max size here?

Reply:
Oh I think when I made this suggestion there was no size. No strong feelings from me :)

Reply:
To be clear -- I've chosen the case of no-max-size, since this is intended to be a pure function called with relatively few arguments, and sometimes on hot code paths.

Reply:
Yep that makes sense. Have updated my suggestion to match that usage model.

Should add the suggestion is included as `cache` was only added in Python 3.9

Reply:
> Should add the suggestion is included as `cache` was only added in Python 3.9

Yeah, I missed that earlier, thought I had learned about a fun old functool today, rather than a fun new one.